sqlglot/optimizer/qualify_columns.py

```diff
@@ -0,0 +1,321 @@
+    for column in columns:
+        column_table = column.text("table")
+
+        if column_table and column_table not in selectables:
+            raise RuntimeError(f"Unknown table reference: {column_table}")
+        if not column_table:
+            if unambiguous_columns is None:
+                unambiguous_columns = _get_unambiguous_columns(selectables)
+
+            column_table = unambiguous_columns.get(column.text("this"))
+            if not column_table:
+                raise RuntimeError(f"Ambiguous column: {column}")
+            column.set("table", exp.to_identifier(column_table))
+
+
+def _qualify_subqueries(subqueries, schema, sequence, selectables):
+    for subquery in subqueries:
+        _qualify_union(subquery, schema, sequence, selectables)
+
+
+def _qualify_outputs(selections):
+    outputs = []
+
+    for i, selection in enumerate(selections):
+        if isinstance(selection, exp.Alias):
+            selection_name = selection.text("alias")
+        elif isinstance(selection, exp.Column):
+            selection_name = selection.text("this")
+            selection.replace(exp.alias_(selection.copy(), selection_name))
+        else:
+            selection_name = f"_col{i}"
+            selection.replace(exp.alias_(selection.copy(), selection_name))
+
+        outputs.append(selection_name)
+
+    return outputs
+
+
+def _merge(*selectables):
+    """
+    Merge two "selectable" dictionaries.
+
+    Raise an error if there are any duplicate selectable names.
+    """
+    result = {}
+    for s in selectables:
+        for selectable_table, selectable_columns in s.items():
+            if selectable_table in selectables:
+                raise RuntimeError(f"Duplicate name declared: {selectable_table}")
+            result[selectable_table] = selectable_columns
+    return result
+
+
+def _get_unambiguous_columns(selectables):
+    """
+    Find all the unambiguous columns in selectables.
+
+    Args:
+        selectables (dict): Mapping of name to selectable columns
+    Returns:
+        dict: Mapping of column name to selectable name
+    """
+    if not selectables:
+        return {}
+
+    selectables = list(selectables.items())
+
+    first_table, first_columns = selectables[0]
+    unambiguous_columns = {col: first_table for col in _unique_columns(first_columns)}
+
+    for table, columns in selectables[1:]:
+        unique = _unique_columns(columns)
+        ambiguous = set(unambiguous_columns).intersection(unique)
+        for column in ambiguous:
+            unambiguous_columns.pop(column)
+        for column in unique.difference(ambiguous):
+            unambiguous_columns[column] = table
+
+    return unambiguous_columns
+
+
+def _unique_columns(columns):
+    """
+    Find the unique columns in a list of columns.
+
+    Example:
+        >>> sorted(_unique_columns(["a", "b", "b", "c"]))
+        ['a', 'c']
+
+    This is necessary because duplicate column names are ambiguous.
+    """
+    unique = set()
```

Comment:
counts = {}
for column in columns:
   counts[column] = counts.get(column, 0) + 1
return [column for column, count in  counts.items() if count == 1]